### PR TITLE
Switch to eslint & drop gulp

### PR DESCRIPTION
**Pull request by tomap**
fix coverage (even if there are no tests)
drop gulp

### CODE REVIEW

**Comment by tomap** [comment]
The coverage is still failing, but I believe it is because of a lack of some coveralls variable
```
$ nyc report --reporter=text-lcov | coveralls
[error] "2019-06-01T18:39:51.781Z"  'error from lcovParse: ' 'Failed to parse string'
[error] "2019-06-01T18:39:51.783Z"  'input: ' ''
[error] "2019-06-01T18:39:51.784Z"  'error from convertLcovToCoveralls'
/home/travis/build/hexojs/hexo-uglify/node_modules/coveralls/bin/coveralls.js:18
        throw err;
        ^
Failed to parse string
```
https://stackoverflow.com/questions/51689696/coverall-module-fails-to-parse-strings

but the rest is much better, isn't it?

**Comment by YoshinoriN** [comment]
@tomap 
This repository was disabled at coveralls.

repo

I have already changed it to enable & re-build travisCI. But, not yet sent to coveralls. I think maybe we will helpful https://github.com/hexojs/hexo/pull/3570

**Comment by weyusi** [comment]
Without test, `nyc report --reporter=text-lcov` would not output anything.

**Comment by tomap** [comment]
I know there are no test, but at least, there is a the starting point to code them, and no more outdated dependencies :)

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ node_modules/
 tmp/
 *.log
 .idea/
-coverage/
\ No newline at end of file
+coverage/
```
Comment (weyusi) [inline .gitignore]: I think this is only used by istanbul?

File: package.json [hunk 2]
```diff
@@ -4,13 +4,15 @@
   "description": "Minify JavaScript files with UglifyJS.",
   "main": "index",
   "scripts": {
-    "test": "gulp test"
+    "test": "nyc mocha",
```
Comment (weyusi) [inline package.json]: `mocha test/index.js` is safer, even though it's unlikely there'll be more than one test files but who knows. `nyc` is only applicable for coverage.

Comment (tomap) [reply package.json]: fixed

**Comment by tomap** [comment]
All comments resolved

**Pull request merged**